gh-99367: Do not mangle sys.path[0] in pdb if safe_path is set (#111762)
python/cpython: Doc/whatsnew/3.13.rst

  identified and executed.
  (Contributed by Tian Gao in :gh:`108464`.)

> * ``sys.path[0]`` will not be mangled if ``sys.flags.safe_path`` is set by
>   :option:`-P` command line option or :envvar:`PYTHONSAFEPATH` environment
>   variable.

Reviewer: ```suggestion
* ``sys.path[0]`` will no longer be mangled when ``sys.flags.safe_path`` is set (via the
  :option:`-P` command line option or :envvar:`PYTHONSAFEPATH` environment
  variable).
```

Reviewer: Is "mangled" clear? Or would it be better two write "will no longer be replaced by..."?

Author: Is it better now? I explained it in detail. Though the users might not even be aware that `sys.path[0]` is replaced by something else, or why.
